src/main/java/com/conveyal/analysis/components/TaskScheduler.java

```diff
@@ -143,39 +134,49 @@ public final void run () {
         }
     }
 
-    ////// Methods for reporting active tasks for each user
+    public Task newTaskForUser (UserPermissions user) {
```

Comment:
This was less thought out overall, but the method exists mainly to abstract away the creation with a `UserPermissions` object (which could be done with a `UserTask`) and add it to a Map based on `UserPermissions` (which, possibly should be `accessGroup` based instead of `email` based, or use both somehow). This way we don't need to know the `UserPermissions` anywhere else.

It was created in conjunction with removing the method `Task.forUser`, which is in the r5 package and probably shouldn't have a dependency on `UserPermissions`.

I was considering creating a separate class, `UserTasks`, which would contain the map of tasks for a specific user, could be it's own component outside of the scheduler. Then the method, `UserTasks.create(...)` would replace the original `Task.forUser` method. `UserTasks` would then also contain the methods discussed elsewhere (`getTasksForUser`, `clearAllCompletedForUser`, `clearTaskForUser`, and `clearOldTasks`). 

It's not 100% wrong for the `Task` to exist in the task list before it's enqueued.

Reply:
As mentioned above, `tasks` may want to key on both `accessGroup` and `user` and therefore have a more complicated implementation that would deserve its own separate class in order to handle `getTasksForAccessGroup` along with `getTasksForUser`.